contracts/protocol/libs/FundsLib.sol

```diff
@@ -207,6 +208,8 @@ library FundsLib {
             payoff.protocol += sequentialProtocolFee;
         }
 
+        handleDRFeeReturn(_exchangeId, exchange, offer, exchangeToken);
```

Comment:
This seems to handle only the cases when fee is returned. 

We also need to handle when DR gets the fee and increase it's available funds. Probably `BosonTypes.Payoff` can be extended to include DR's payoff to and follow the pattern for others.